src/VisualStudio/Core/Def/Implementation/Library/FindResults/TreeItems/MetadataDefinitionTreeItem.cs

```diff
@@ -36,7 +35,17 @@ public MetadataDefinitionTreeItem(ISymbol definition, ushort glyphIndex)
 
         public override int GoToSource()
         {
-            return VSConstants.E_NOTIMPL;
+            var resolution = _symbolKey.Resolve(_workspace.CurrentSolution.GetCompilationAsync(_referencingProjectId, CancellationToken.None).Result);
```

Comment:
@balajikris David isn't comparing SymbolKeys that came from different compilations. He's resolving a SymbolKey using Compilations from the same project, so there may actually be corner cases where you want to use assembly identity. (Imagine using FAR, getting a metadata reference, and then changing a reference such that there was a type with the same SymbolKey in a different assembly. Probably wouldn't want to navigate to it's MAS).